### PR TITLE
store estimated reading time in database (#393)

### PR DESCRIPTION
**Pull request by nicosomb**
see #393

### CODE REVIEW

**Comment by tcitworld** [comment]
Any idea what's happening with Travis CI ?

**Comment by tcitworld** [comment]
I wonder if we should try looking for another solution. This method to estimate reading time is kind of *really* bad with non-Latin characters and/or some languages.

**Comment by nicosomb** [comment]
What do you suggest?

**Comment by tcitworld** [comment]
> try looking for another solution

;-)

**Comment by nicosomb** [comment]
@j0k3r I really don't understand why tests failed. Maybe I'm so tired. Go to bed. If you can help me ... ;-)

**Comment by tcitworld** [comment]
Didn't find any real alternative, but it seems to me we could take in consideration pictures, with a time related to their size.

**Comment by nicosomb** [comment]
Yeah, I found my error. Fixed.

**Comment by nicosomb** [comment]
> but it seems to me we could take in consideration pictures

Seriously? It's an estimation of reading time, I don't think we should take in consideration pictures.

**Comment by tcitworld** [comment]
Well, I thought it more like the time spent on the whole article. Weren't we considering counting the length of videos too ?

File: src/Wallabag/CoreBundle/Helper/Tools.php [hunk 1]
```diff
@@ -118,4 +118,16 @@ public static function generateToken()
...
+    /**
+     * For a given text, we calculate reading time for an article.
+     *
+     * @param $text
+     *
+     * @return float
+     */
+    public static function getReadingTime($text)
+    {
+        return floor(str_word_count(strip_tags($text)) / 200);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Helper/Tools.php]: WP plugin seems to use 250 words per minutes ([see here](https://plugins.trac.wordpress.org/browser/estimated-post-reading-time/trunk/plugin.php#L73)), maybe we can adjust it ?
Also, it could be interesting to explain the calculation, at least what is this number `200`.

**Comment by j0k3r** [comment]
I don't this a lot of tests in this PR :)

**Comment by tcitworld** [comment]
Wikipedia shows some data about the number of words reads per minute : https://en.m.wikipedia.org/wiki/Reading_(process)#Reading_rate

**Comment by nicosomb** [comment]
@modos189 I'll add filters to filter articles by estimated reading view. I'll add a select list (with 3 items "< 5min", "between 5 and 10" and "> 10 min") and 2 text fields "min" and "max" to allow the user to filter with personal filters. 
If my explanations are clear, can you imagine how we can implement that on Unread / starred / archive screen?

Next, we'll have a new filter, by domain name.

**Comment by modos189** [comment]
screen6

**Comment by nicosomb** [comment]
I'm not sure that the filters icon is obvious for everyone.
And do you think popin is good for users? 
Do you think we can consider a right sidebar? 

Thank you for your reactivity.

**Comment by tcitworld** [comment]
It's normal when using material design, but on an desktop view it's taking time to access.

**Comment by tcitworld** [comment]
Also, maybe buttons instead of a list to choose from (one click instead of two).

**Comment by nicosomb** [comment]
> It's normal when using material design
You talk about which point?

**Comment by tcitworld** [comment]
The filter icon. You find it on any android app using material design.

**Comment by nicosomb** [comment]
> I don't this a lot of tests in this PR :) 
This PR is still in progress, don't worry ;-)

**Comment by modos189** [comment]
screen7
I do not want the icon to change

**Comment by tcitworld** [comment]
I find this nice.

**Comment by nicosomb** [comment]
:+1:

**Comment by tcitworld** [comment]
And we can add more filter options in the future.

**Comment by tcitworld** [comment]
As for more filters (I write it here, I know it's dirty), there should be the date when the entry was created and the type (MIME type in database).

**Comment by nicosomb** [comment]
@j0k3r how do you think we can implement tests for this feature? Just by adding some articles (and setting estimating time by hand) and filling filter form?

File: src/Wallabag/CoreBundle/Controller/EntryController.php [hunk 2]
```diff
@@ -32,6 +36,7 @@ public function addEntryAction(Request $request)
 
             $entry->setTitle($content->getTitle());
             $entry->setContent($content->getBody());
+            $entry->setReadingTime(Tools::getReadingTime($content->getBody()));
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: Why don't you do that directly in the EntryEntity ?

File: src/Wallabag/CoreBundle/Controller/EntryController.php [hunk 3]
```diff
@@ -89,22 +94,39 @@ public function editEntryAction(Request $request, Entry $entry)
...
      */
-    public function showUnreadAction($page)
+    public function showUnreadAction(Request $request, $page)
     {
-        $entries = $this->getDoctrine()
+        $form = $this->get('form.factory')->create(new EntryFilterType());
+
+        $filterBuilder = $this->getDoctrine()
             ->getRepository('WallabagCoreBundle:Entry')
             ->findUnreadByUser($this->getUser()->getId());
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: Did you really need to call it `find`? `getFilterForUnreadByUser` doesn't work?

**Comment by j0k3r** [comment]
Just defined some fixtures (and/or updated existing one) and they will have estimated time. Then create a test for a filter by filling the form and check number of results.

File: src/Wallabag/CoreBundle/Controller/EntryController.php (hunk 3 above)
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: It must start by findBy or findOne...

File: src/Wallabag/CoreBundle/Controller/EntryController.php (hunk 2 above)
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Controller/EntryController.php]: fixed.

**Comment by nicosomb** [comment]
I pushed my draft for test. I need help for this one, if you can have a look...

**Comment by nicosomb** [comment]
Ready for review.

File: src/Wallabag/CoreBundle/Repository/EntryRepository.php [hunk 4]
```diff
@@ -17,16 +17,11 @@ class EntryRepository extends EntityRepository
...
     {
-        $qb = $this->createQueryBuilder('e')
+        return $this->createQueryBuilder('e')
             ->leftJoin('e.user', 'u')
             ->where('e.isArchived = false')
             ->andWhere('u.id =:userId')->setParameter('userId', $userId)
-            ->orderBy('e.id', 'desc')
-            ->getQuery();
-
-        $pagerAdapter = new DoctrineORMAdapter($qb);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Repository/EntryRepository.php]: Sine you removed lines where `DoctrineORMAdapter` is used, you can remove the `use` at the beginning of the file

File: src/Wallabag/CoreBundle/Helper/Tools.php (hunk 1 above)
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Helper/Tools.php]: Could you just add a comment about the 200 number? Telling that it is referencing to _words per minute_

File: src/Wallabag/CoreBundle/Resources/views/Entry/entries.html.twig [hunk 5]
```diff
@@ -21,11 +21,12 @@
     {% if entries is empty %}
         <div class="messages warning"><p>{% trans %}No articles found.{% endtrans %}</p></div>
     {% else %}
+        <div><form>{{ form_rest(form) }}<button class="btn waves-effect waves-light" type="submit" id="submit-filter" value="filter">Filter</button></form></div>
         {% for entry in entries %}
             <div id="entry-{{ entry.id|e }}" class="entrie">
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/Entry/entries.html.twig]: `entrie`, really? It's not english at all, it should be entry.
Maybe it should be done in a different PR, but you are using this class in your test, some maybe you can fix it

Comment (nicosomb) [inline src/Wallabag/CoreBundle/Resources/views/Entry/entries.html.twig]: it's baggy theme ... I don't know if I have to fix this theme here...

Comment (tcitworld) [inline src/Wallabag/CoreBundle/Resources/views/Entry/entries.html.twig]: I started migration work on the baggy theme this afternoon, but you can fix this here, no worries.

File: src/Wallabag/CoreBundle/Repository/EntryRepository.php (hunk 4 above)
Comment (nicosomb) [inline src/Wallabag/CoreBundle/Repository/EntryRepository.php]: still used at the end of the file.

**Comment by nicosomb** [comment]
finally good to merge?

File: src/Wallabag/CoreBundle/Repository/EntryRepository.php (hunk 4 above)
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Repository/EntryRepository.php]: Woops, sorry
